### ISSUE TITLE
briefcase run android gives error: "No build receipt resource found"

### ISSUE THREAD

**Issue by mhsmith** [beeware/briefcase#1060]
_Originally posted by @CarlosHGL in https://github.com/beeware/briefcase/issues/1017#issuecomment-1401311140_

Now, I am having an error during the building of the briefcase for the android app, but it works on windows. The app is completely standard, nothing was changed. All I did was create the virtual environment, create the app through the new briefcase, and run briefcase run android. And, boom! It gives me the same annoying errors that it gave me in all my other attempts.

Terminal feedback
```
(env) PS C:\Projetos\Projetos\Remenber Me!\noname> briefcase run android

[noname] Building Android APK...
Exception in thread "main" java.lang.RuntimeException: Could not determine wrapper version.
        at org.gradle.wrapper.GradleWrapperMain.wrapperVersion(GradleWrapperMain.java:111)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
Caused by: java.lang.RuntimeException: No build receipt resource found.
        at org.gradle.wrapper.GradleWrapperMain.wrapperVersion(GradleWrapperMain.java:97)
        ... 1 more
Building...

Error while building project.
```

**Logs**
```Date/Time:      2023-01-23 22:01:38 
Command line:    C:\Projetos\Projetos\Remenber Me!\env\Scripts\briefcase run android

OS Release:      Windows 10
OS Version:      10.0.22623
Architecture:    AMD64
Platform:        Windows-10-10.0.22623-SP0

Python exe:      C:\Projetos\Projetos\Remenber Me!\env\Scripts\python.exe
Python version:  3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
Virtual env:     True
Conda env:       False

Briefcase:       0.3.11
Target platform: android
Target format:   gradle

Environment Variables:
    ALLUSERSPROFILE=C:\ProgramData
    APPDATA=C:\Users\Positivo\AppData\Roaming
    CHROME_CRASHPAD_PIPE_NAME=\\.\pipe\LOCAL\crashpad_6116_QNFURFTEALBAWLLC
    COLORTERM=truecolor
    COMMONPROGRAMFILES=C:\Program Files\Common Files
    COMMONPROGRAMFILES(X86)=C:\Program Files (x86)\Common Files
    COMMONPROGRAMW6432=C:\Program Files\Common Files
    COMPUTERNAME=COMPUTADOR-CARL
    COMSPEC=C:\WINDOWS\system32\cmd.exe
    DRIVERDATA=C:\Windows\System32\Drivers\DriverData
    FPS_BROWSER_APP_PROFILE_STRING=Internet Explorer
    FPS_BROWSER_USER_PROFILE_STRING=Default
    GIT_ASKPASS=********************
    HOMEDRIVE=C:
    HOMEPATH=\Users\Positivo
    LANG=pt_BR.UTF-8
    LOCALAPPDATA=C:\Users\Positivo\AppData\Local
    LOGONSERVER=\\COMPUTADOR-CARL
    NUMBER_OF_PROCESSORS=2
    ONEDRIVE=C:\Users\Positivo\OneDrive
    ONEDRIVECONSUMER=C:\Users\Positivo\OneDrive
    ORIGINAL_XDG_CURRENT_DESKTOP=undefined
    OS=Windows_NT
    PATH=C:\Projetos\Projetos\Remenber Me!\env\Scripts;C:\Program Files\PowerShell\7;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Users\Positivo\AppData\Local\Programs\Python\Python38\Scripts;C:\Program Files\dotnet\;C:\Program Files\PuTTY\;C:\Program Files\Git\cmd;C:\Program Files\PowerShell\7\;C:\Program Files\nodejs\;C:\Users\Positivo\AppData\Local\Programs\Python\Python310\Scripts\;C:\Users\Positivo\AppData\Local\Programs\Python\Python310\;C:\Program Files\MySQL\MySQL Shell 8.0\bin\;C:\Users\Positivo\scoop\shims;C:\Users\Positivo\AppData\Local\Microsoft\WindowsApps;;C:\Program Files\JetBrains\PyCharm Community Edition 2021.1.2\bin;C:\Users\Positivo\AppData\Local\Microsoft\WindowsApps;C:\Program Files\heroku\bin;C:\Users\Positivo\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Positivo\AppData\Roaming\npm
    PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL
    POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows 10 Home Single Language
    PROCESSOR_ARCHITECTURE=AMD64
    PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 76 Stepping 4, GenuineIntel
    PROCESSOR_LEVEL=6
    PROCESSOR_REVISION=4c04
    PROGRAMDATA=C:\ProgramData
    PROGRAMFILES=C:\Program Files
    PROGRAMFILES(X86)=C:\Program Files (x86)
    PROGRAMW6432=C:\Program Files
    PSMODULEPATH=C:\Users\Positivo\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
    PUBLIC=C:\Users\Public
    PYCHARM COMMUNITY EDITION=C:\Program Files\JetBrains\PyCharm Community Edition 2021.1.2\bin;
    PYTHONUSERBASE=C:\Users\Positivo\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages
    SESSIONNAME=Console
    SYSTEMDRIVE=C:
    SYSTEMROOT=C:\WINDOWS
    TEMP=C:\Users\Positivo\AppData\Local\Temp
    TERM_PROGRAM=vscode
    TERM_PROGRAM_VERSION=1.74.3
    TMP=C:\Users\Positivo\AppData\Local\Temp
    USERDOMAIN=COMPUTADOR-CARL
    USERDOMAIN_ROAMINGPROFILE=COMPUTADOR-CARL
    USERNAME=Positivo
    USERPROFILE=C:\Users\Positivo
    VIRTUAL_ENV=C:\Projetos\Projetos\Remenber Me!\env
    VIRTUAL_ENV_PROMPT=env
    VSCODE_GIT_ASKPASS_EXTRA_ARGS=********************
    VSCODE_GIT_ASKPASS_MAIN=********************
    VSCODE_GIT_ASKPASS_NODE=********************
    VSCODE_GIT_IPC_HANDLE=\\.\pipe\vscode-git-5c29dc16bb-sock
    VSCODE_INJECTION=1
    WINDIR=C:\WINDOWS
    _OLD_VIRTUAL_PATH=C:\Program Files\PowerShell\7;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Users\Positivo\AppData\Local\Programs\Python\Python38\Scripts;C:\Program Files\dotnet\;C:\Program Files\PuTTY\;C:\Program Files\Git\cmd;C:\Program Files\PowerShell\7\;C:\Program Files\nodejs\;C:\Users\Positivo\AppData\Local\Programs\Python\Python310\Scripts\;C:\Users\Positivo\AppData\Local\Programs\Python\Python310\;C:\Program Files\MySQL\MySQL Shell 8.0\bin\;C:\Users\Positivo\scoop\shims;C:\Users\Positivo\AppData\Local\Microsoft\WindowsApps;;C:\Program Files\JetBrains\PyCharm Community Edition 2021.1.2\bin;C:\Users\Positivo\AppData\Local\Microsoft\WindowsApps;C:\Program Files\heroku\bin;C:\Users\Positivo\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Positivo\AppData\Roaming\npm

Briefcase Log:
[22:01:30] >>> Android emulator is already installed.                                                                                                             android_sdk.py:409
                                                                                                                                                                       gradle.py:149
           [noname] Building Android APK...                                                                                                                            gradle.py:149
                                                                                                                                                                   subprocess.py:516
           >>> Running Command:                                                                                                                                    subprocess.py:517
           >>>     'C:\Projetos\Projetos\Remenber Me!\noname\android\gradle\No Name\gradlew.bat' assembleDebug --console plain                                     subprocess.py:518
           >>> Working Directory:                                                                                                                                  subprocess.py:525
           >>>     C:\Projetos\Projetos\Remenber Me!\noname\android\gradle\No Name                                                                                 subprocess.py:526
           >>> Environment Overrides:                                                                                                                              subprocess.py:535
           >>>     ANDROID_SDK_ROOT=C:\Users\Positivo\AppData\Local\BeeWare\briefcase\Cache\tools\android_sdk                                                      subprocess.py:537
           >>>     JAVA_HOME=C:\Users\Positivo\AppData\Local\BeeWare\briefcase\Cache\tools\java                                                                    subprocess.py:537
[22:01:31] >>> Return code: 1                                                                                                                                      subprocess.py:553
           Building...                                                                                                                                                 gradle.py:150
                                                                                                                                                                      __main__.py:27
           Error while building project.                                                                                                                              __main__.py:28

╭─────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ────────────────────────────────────────────────────────────────────────╮
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\platforms\android\gradle.py:152 in build_app                                                                   │
│                                                                                                                                                                                  │
│   149 │   │   self.logger.info("Building Android APK...", prefix=app.app_name)                                                                                                   │
│   150 │   │   with self.input.wait_bar("Building..."):                                                                                                                           │
│   151 │   │   │   try:                                                                                                                                                           │
│ ❱ 152 │   │   │   │   self.tools.subprocess.run(                                                                                                                                 │
│   153 │   │   │   │   │   # Windows needs the full path to `gradlew`; macOS & Linux can find i                                                                                   │
│   154 │   │   │   │   │   # via `./gradlew`. For simplicity of implementation, we always provi                                                                                   │
│   155 │   │   │   │   │   # the full path.                                                                                                                                       │
│                                                                                                                                                                                  │
│ ╭─────────────────────────────────────────── locals ────────────────────────────────────────────╮                                                                                │
│ │    app = <online.carloslima.noname v0.0.1 AppConfig>                                          │                                                                                │
│ │ kwargs = {'device_or_avd': None}                                                              │                                                                                │
│ │   self = <briefcase.platforms.android.gradle.GradleBuildCommand object at 0x0000018DF2E49D20> │                                                                                │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────╯                                                                                │
│                                                                                                                                                                                  │
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\integrations\subprocess.py:125 in inner                                                                        │
│                                                                                                                                                                                  │
│   122 │   │   # Run subprocess command with or without console control                                                                                                           │
│   123 │   │   if remove_dynamic_elements:                                                                                                                                        │
│   124 │   │   │   with sub.tools.input.release_console_control():                                                                                                                │
│ ❱ 125 │   │   │   │   return sub_method(sub, args, **kwargs)                                                                                                                     │
│   126 │   │   else:                                                                                                                                                              │
│   127 │   │   │   return sub_method(sub, args, **kwargs)                                                                                                                         │
│   128                                                                                                                                                                            │
│                                                                                                                                                                                  │
│ ╭─────────────────────────────────────────────────────────────────────────── locals ────────────────────────────────────────────────────────────────────────────╮                │
│ │                    args = [WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name/gradlew.bat'), 'assembleDebug', '--console', 'plain'] │                │
│ │              executable = 'C:\\Projetos\\Projetos\\Remenber Me!\\noname\\android\\gradle\\No Name\\gradlew.bat'                                               │                │
│ │                  kwargs = {                                                                                                                                   │                │
│ │                           │   'env': {                                                                                                                        │                │
│ │                           │   │   'ANDROID_SDK_ROOT': 'C:\\Users\\Positivo\\AppData\\Local\\BeeWare\\briefcase\\Cache\\tools\\android_sdk',                   │                │
│ │                           │   │   'JAVA_HOME': 'C:\\Users\\Positivo\\AppData\\Local\\BeeWare\\briefcase\\Cache\\tools\\java'                                  │                │
│ │                           │   },                                                                                                                              │                │
│ │                           │   'cwd': WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name'),                                          │                │
│ │                           │   'check': True                                                                                                                   │                │
│ │                           }                                                                                                                                   │                │
│ │ remove_dynamic_elements = True                                                                                                                                │                │
│ │                     sub = <briefcase.integrations.subprocess.Subprocess object at 0x0000018DF2E49BA0>                                                         │                │
│ │              sub_method = <function Subprocess.run at 0x0000018DF2D7E950>                                                                                     │                │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                │
│                                                                                                                                                                                  │
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\integrations\subprocess.py:292 in run                                                                          │
│                                                                                                                                                                                  │
│   289 │   │   self._log_environment(kwargs.get("env"))                                                                                                                           │
│   290 │   │                                                                                                                                                                      │
│   291 │   │   try:                                                                                                                                                               │
│ ❱ 292 │   │   │   command_result = self._subprocess.run(                                                                                                                         │
│   293 │   │   │   │   [str(arg) for arg in args], **self.final_kwargs(**kwargs)                                                                                                  │
│   294 │   │   │   )                                                                                                                                                              │
│   295 │   │   except subprocess.CalledProcessError as e:                                                                                                                         │
│                                                                                                                                                                                  │
│ ╭────────────────────────────────────────────────────────────────────────── locals ──────────────────────────────────────────────────────────────────────────╮                   │
│ │                 args = [WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name/gradlew.bat'), 'assembleDebug', '--console', 'plain'] │                   │
│ │ is_output_redirected = None                                                                                                                                │                   │
│ │               kwargs = {                                                                                                                                   │                   │
│ │                        │   'env': {                                                                                                                        │                   │
│ │                        │   │   'ANDROID_SDK_ROOT': 'C:\\Users\\Positivo\\AppData\\Local\\BeeWare\\briefcase\\Cache\\tools\\android_sdk',                   │                   │
│ │                        │   │   'JAVA_HOME': 'C:\\Users\\Positivo\\AppData\\Local\\BeeWare\\briefcase\\Cache\\tools\\java'                                  │                   │
│ │                        │   },                                                                                                                              │                   │
│ │                        │   'cwd': WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name'),                                          │                   │
│ │                        │   'check': True                                                                                                                   │                   │
│ │                        }                                                                                                                                   │                   │
│ │                 self = <briefcase.integrations.subprocess.Subprocess object at 0x0000018DF2E49BA0>                                                         │                   │
│ │        stream_output = False                                                                                                                               │                   │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                   │
│                                                                                                                                                                                  │
│ C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.1264.0_x64__qbz5n2kfra8p0\lib\subprocess.py:524 in run                                                    │
│                                                                                                                                                                                  │
│    521 │   │   │   raise                                                                                                                                                         │
│    522 │   │   retcode = process.poll()                                                                                                                                          │
│    523 │   │   if check and retcode:                                                                                                                                             │
│ ❱  524 │   │   │   raise CalledProcessError(retcode, process.args,                                                                                                               │
│    525 │   │   │   │   │   │   │   │   │    output=stdout, stderr=stderr)                                                                                                        │
│    526 │   return CompletedProcess(process.args, retcode, stdout, stderr)                                                                                                        │
│    527                                                                                                                                                                           │
│                                                                                                                                                                                  │
│ ╭────────────────────────────────────────────────────────────────────── locals ──────────────────────────────────────────────────────────────────────╮                           │
│ │ capture_output = False                                                                                                                             │                           │
│ │          check = True                                                                                                                              │                           │
│ │          input = None                                                                                                                              │                           │
│ │         kwargs = {                                                                                                                                 │                           │
│ │                  │   'env': {                                                                                                                      │                           │
│ │                  │   │   'ALLUSERSPROFILE': 'C:\\ProgramData',                                                                                     │                           │
│ │                  │   │   'APPDATA': 'C:\\Users\\Positivo\\AppData\\Roaming',                                                                       │                           │
│ │                  │   │   'CHROME_CRASHPAD_PIPE_NAME': '\\\\.\\pipe\\LOCAL\\crashpad_6116_QNFURFTEALBAWLLC',                                        │                           │
│ │                  │   │   'COMMONPROGRAMFILES': 'C:\\Program Files\\Common Files',                                                                  │                           │
│ │                  │   │   'COMMONPROGRAMFILES(X86)': 'C:\\Program Files (x86)\\Common Files',                                                       │                           │
│ │                  │   │   'COMMONPROGRAMW6432': 'C:\\Program Files\\Common Files',                                                                  │                           │
│ │                  │   │   'COMPUTERNAME': 'COMPUTADOR-CARL',                                                                                        │                           │
│ │                  │   │   'COMSPEC': 'C:\\WINDOWS\\system32\\cmd.exe',                                                                              │                           │
│ │                  │   │   'DRIVERDATA': 'C:\\Windows\\System32\\Drivers\\DriverData',                                                               │                           │
│ │                  │   │   'FPS_BROWSER_APP_PROFILE_STRING': 'Internet Explorer',                                                                    │                           │
│ │                  │   │   ... +50                                                                                                                   │                           │
│ │                  │   },                                                                                                                            │                           │
│ │                  │   'cwd': 'C:\\Projetos\\Projetos\\Remenber Me!\\noname\\android\\gradle\\No Name',                                              │                           │
│ │                  │   'text': True,                                                                                                                 │                           │
│ │                  │   'encoding': 'cp850'                                                                                                           │                           │
│ │                  }                                                                                                                                 │                           │
│ │      popenargs = (['C:\\Projetos\\Projetos\\Remenber Me!\\noname\\android\\gradle\\No Name\\gradlew.bat', 'assembleDebug', '--console', 'plain'],) │                           │
│ │        process = <Popen: returncode: 1 args: ['C:\\Projetos\\Projetos\\Remenber Me!\\noname\\...>                                                  │                           │
│ │        retcode = 1                                                                                                                                 │                           │
│ │         stderr = None                                                                                                                              │                           │
│ │         stdout = None                                                                                                                              │                           │
│ │        timeout = None                                                                                                                              │                           │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
CalledProcessError: Command '['C:\\Projetos\\Projetos\\Remenber Me!\\noname\\android\\gradle\\No Name\\gradlew.bat', 'assembleDebug', '--console', 'plain']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

╭─────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ────────────────────────────────────────────────────────────────────────╮
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\__main__.py:21 in main                                                                                         │
│                                                                                                                                                                                  │
│   18 │   │   options = command.parse_options(extra=extra_cmdline)                                                                                                                │
│   19 │   │   command.check_obsolete_data_dir()                                                                                                                                   │
│   20 │   │   command.parse_config(Path.cwd() / "pyproject.toml")                                                                                                                 │
│ ❱ 21 │   │   command(**options)                                                                                                                                                  │
│   22 │   except HelpText as e:                                                                                                                                                   │
│   23 │   │   logger.info()                                                                                                                                                       │
│   24 │   │   logger.info(str(e))                                                                                                                                                 │
│                                                                                                                                                                                  │
│ ╭────────────────────────────────────────────── locals ──────────────────────────────────────────────╮                                                                           │
│ │       command = <briefcase.platforms.android.gradle.GradleRunCommand object at 0x0000018DF0BCECE0> │                                                                           │
│ │       Command = <class 'briefcase.platforms.android.gradle.GradleRunCommand'>                      │                                                                           │
│ │       console = <briefcase.console.Console object at 0x0000018DF06BC220>                           │                                                                           │
│ │             e = BriefcaseCommandError('Error while building project.')                             │                                                                           │
│ │ extra_cmdline = []                                                                                 │                                                                           │
│ │        logger = <briefcase.console.Log object at 0x0000018DF06BC2B0>                               │                                                                           │
│ │       options = {'appname': None, 'update': False, 'device_or_avd': None}                          │                                                                           │
│ │        result = 200                                                                                │                                                                           │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────╯                                                                           │
│                                                                                                                                                                                  │
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\commands\run.py:62 in __call__                                                                                 │
│                                                                                                                                                                                  │
│   59 │   │   │   state = self.update_command(app, **options)                                                                                                                     │
│   60 │   │   │   state = self.build_command(app, **full_options(state, options))                                                                                                 │
│   61 │   │   elif not binary_file.exists():                                                                                                                                      │
│ ❱ 62 │   │   │   state = self.build_command(app, **options)                                                                                                                      │
│   63 │   │   else:                                                                                                                                                               │
│   64 │   │   │   state = None                                                                                                                                                    │
│   65                                                                                                                                                                             │
│                                                                                                                                                                                  │
│ ╭───────────────────────────────────────────────────────────────── locals ─────────────────────────────────────────────────────────────────╮                                     │
│ │           app = <online.carloslima.noname v0.0.1 AppConfig>                                                                              │                                     │
│ │       appname = None                                                                                                                     │                                     │
│ │   binary_file = WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name/app/build/outputs/apk/debug/app-debug.apk') │                                     │
│ │       options = {'device_or_avd': None}                                                                                                  │                                     │
│ │          self = <briefcase.platforms.android.gradle.GradleRunCommand object at 0x0000018DF0BCECE0>                                       │                                     │
│ │ template_file = WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name')                                           │                                     │
│ │        update = False                                                                                                                    │                                     │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯                                     │
│                                                                                                                                                                                  │
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\commands\build.py:56 in __call__                                                                               │
│                                                                                                                                                                                  │
│   53 │   │   self.verify_tools()                                                                                                                                                 │
│   54 │   │                                                                                                                                                                       │
│   55 │   │   if app:                                                                                                                                                             │
│ ❱ 56 │   │   │   state = self._build_app(app, update=update, **options)                                                                                                          │
│   57 │   │   else:                                                                                                                                                               │
│   58 │   │   │   state = None                                                                                                                                                    │
│   59 │   │   │   for app_name, app in sorted(self.apps.items()):                                                                                                                 │
│                                                                                                                                                                                  │
│ ╭──────────────────────────────────────────── locals ────────────────────────────────────────────╮                                                                               │
│ │     app = <online.carloslima.noname v0.0.1 AppConfig>                                          │                                                                               │
│ │ options = {'device_or_avd': None}                                                              │                                                                               │
│ │    self = <briefcase.platforms.android.gradle.GradleBuildCommand object at 0x0000018DF2E49D20> │                                                                               │
│ │  update = False                                                                                │                                                                               │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────╯                                                                               │
│                                                                                                                                                                                  │
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\commands\build.py:41 in _build_app                                                                             │
│                                                                                                                                                                                  │
│   38 │   │                                                                                                                                                                       │
│   39 │   │   self.verify_app_tools(app)                                                                                                                                          │
│   40 │   │                                                                                                                                                                       │
│ ❱ 41 │   │   state = self.build_app(app, **full_options(state, options))                                                                                                         │
│   42 │   │                                                                                                                                                                       │
│   43 │   │   self.logger.info(                                                                                                                                                   │
│   44 │   │   │   f"Built {self.binary_path(app).relative_to(self.base_path)}",                                                                                                   │
│                                                                                                                                                                                  │
│ ╭────────────────────────────────────────────── locals ──────────────────────────────────────────────╮                                                                           │
│ │         app = <online.carloslima.noname v0.0.1 AppConfig>                                          │                                                                           │
│ │     options = {'device_or_avd': None}                                                              │                                                                           │
│ │        self = <briefcase.platforms.android.gradle.GradleBuildCommand object at 0x0000018DF2E49D20> │                                                                           │
│ │       state = None                                                                                 │                                                                           │
│ │ target_file = WindowsPath('C:/Projetos/Projetos/Remenber Me!/noname/android/gradle/No Name')       │                                                                           │
│ │      update = False                                                                                │                                                                           │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────╯                                                                           │
│                                                                                                                                                                                  │
│ C:\Projetos\Projetos\Remenber Me!\env\lib\site-packages\briefcase\platforms\android\gradle.py:164 in build_app                                                                   │
│                                                                                                                                                                                  │
│   161 │   │   │   │   │   check=True,                                                                                                                                            │
│   162 │   │   │   │   )                                                                                                                                                          │
│   163 │   │   │   except subprocess.CalledProcessError as e:                                                                                                                     │
│ ❱ 164 │   │   │   │   raise BriefcaseCommandError("Error while building project.") from e                                                                                        │
│   165                                                                                                                                                                            │
│   166                                                                                                                                                                            │
│   167 class GradleRunCommand(GradleMixin, RunCommand):                                                                                                                           │
│                                                                                                                                                                                  │
│ ╭─────────────────────────────────────────── locals ────────────────────────────────────────────╮                                                                                │
│ │    app = <online.carloslima.noname v0.0.1 AppConfig>                                          │                                                                                │
│ │ kwargs = {'device_or_avd': None}                                                              │                                                                                │
│ │   self = <briefcase.platforms.android.gradle.GradleBuildCommand object at 0x0000018DF2E49D20> │                                                                                │
│ ╰───────────────────────────────────────────────────────────────────────────────────────────────╯                                                                                │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
BriefcaseCommandError: Error while building project.
[22:01:32]                                                                                                                                                            console.py:269
           >>> Extra information:                                                                                                                                     console.py:270
                                                                                                                                                                   subprocess.py:516
           >>> Running Command:                                                                                                                                    subprocess.py:517
[22:01:33] >>>     'C:\Users\Positivo\AppData\Local\BeeWare\briefcase\Cache\tools\android_sdk\cmdline-tools\latest\bin\sdkmanager.bat' --list_installed            subprocess.py:518
           >>> Working Directory:                                                                                                                                  subprocess.py:525
           >>>     C:\Projetos\Projetos\Remenber Me!\noname                                                                                                        subprocess.py:526
           >>> Environment Overrides:                                                                                                                              subprocess.py:535
           >>>     ANDROID_SDK_ROOT=C:\Users\Positivo\AppData\Local\BeeWare\briefcase\Cache\tools\android_sdk                                                      subprocess.py:537
           >>>     JAVA_HOME=C:\Users\Positivo\AppData\Local\BeeWare\briefcase\Cache\tools\java                                                                    subprocess.py:537
[22:01:38] >>> Command Output:                                                                                                                                     subprocess.py:542
           >>>     Loading package information...                                                                                                                  subprocess.py:544
           >>>     Loading local repository...                                                                                                                     subprocess.py:544
           >>>     [=========                              ] 25% Loading local repository...                                                                       subprocess.py:544
           >>>     [=========                              ] 25% Fetch remote repository...                                                                        subprocess.py:544
           >>>     [=======================================] 100% Fetch remote repository...                                                                       subprocess.py:544
           >>>     Installed packages:                                                                                                                             subprocess.py:544
           >>>       Path                 | Version      | Description                     | Location                                                              subprocess.py:544
           >>>       -------              | -------      | -------                         | -------                                                               subprocess.py:544
           >>>       build-tools;30.0.2   | 30.0.2       | Android SDK Build-Tools 30.0.2  | build-tools\30.0.2                                                    subprocess.py:544
           >>>       cmake;3.10.2.4988404 | 3.10.2       | CMake 3.10.2.4988404            | cmake\3.10.2.4988404                                                  subprocess.py:544
           >>>       emulator             | 31.3.14      | Android Emulator                | emulator                                                              subprocess.py:544
           >>>       ndk;21.4.7075529     | 21.4.7075529 | NDK (Side by side) 21.4.7075529 | ndk\21.4.7075529                                                      subprocess.py:544
           >>>       patcher;v4           | 1            | SDK Patch Applier v4            | patcher\v4                                                            subprocess.py:544
           >>>       platform-tools       | 33.0.3       | Android SDK Platform-Tools      | platform-tools                                                        subprocess.py:544
           >>>       platforms;android-32 | 1            | Android SDK Platform 32         | platforms\android-32                                                  subprocess.py:544
           >>>                                                                                                                                                     subprocess.py:544
           >>> Return code: 0                                                                                                                                      subprocess.py:553```

**Comment by mhsmith** [comment]
If you Google for "No build receipt resource found", the first result is https://stackoverflow.com/questions/41560235, which says this is caused by having a `!` character in your project directory name.

**Comment by mhsmith** [comment]
Please rename it and try again. If you get the same error, please post a comment here. If you get a different error, please create a separate issue.

**Comment by mhsmith** [comment]
@rmartin16: It looks like this is another case of an error message appearing on the terminal but not in the log file. Will this be fixed by #1059?

**Comment by rmartin16** [comment]
> @rmartin16: It looks like this is another case of an error message appearing on the terminal but not in the log file. Will this be fixed by #1059?

Ahh...this is the case of output from batch scripts being absent from the log. This was resolved in #943.

**Comment by CarlosHGL** [comment]
The problem has been resolved, thank you. The problem, for some reason, was being caused by the python from the Windows store, even though I wasn't using it to create my virtual environment, it was interfering with the compilation of my application. So, I simply uninstalled it from my computer and everything worked as it should. Thank you.
